tests-utils/src/main/scala/tech/beshu/ror/utils/containers/images/Elasticsearch.scala

```diff
@@ -35,28 +38,92 @@ object Elasticsearch {
 
   extension (config: Config)
     def esConfigDir: Path = config.esInstallationType match {
-      case EsInstallationType.EsDockerImage => os.root / "usr" / "share" / "elasticsearch" / "config"
-      case EsInstallationType.UbuntuDockerImageWithEsFromApt => os.root / "etc" / "elasticsearch"
+      case EsInstallationType.EsDockerImage =>
+        os.root / "usr" / "share" / "elasticsearch" / "config"
+      case EsInstallationType.UbuntuDockerImageWithEsFromApt =>
+        os.root / "etc" / "elasticsearch"
+      case EsInstallationType.NativeWindowsProcess =>
+        WindowsEsDirectoryManager.configPath(config.clusterName, config.nodeName)
+    }
+
+    def esDir: Path = config.esInstallationType match {
+      case EsInstallationType.EsDockerImage =>
+        os.root / "usr" / "share" / "elasticsearch"
+      case EsInstallationType.UbuntuDockerImageWithEsFromApt =>
+        os.root / "usr" / "share" / "elasticsearch"
+      case EsInstallationType.NativeWindowsProcess =>
+        WindowsEsDirectoryManager.esPath(config.clusterName, config.nodeName)
+    }
+
+    def tempFilePath: Path = config.esInstallationType match {
+      case EsInstallationType.EsDockerImage =>
+        os.root / "tmp"
+      case EsInstallationType.UbuntuDockerImageWithEsFromApt =>
+        os.root / "tmp"
+      case EsInstallationType.NativeWindowsProcess =>
+        WindowsEsDirectoryManager.esPath(config.clusterName, config.nodeName) / "temp"
     }
 
   sealed trait EsInstallationType
 
   object EsInstallationType {
+    case object NativeWindowsProcess extends EsInstallationType
+
     case object EsDockerImage extends EsInstallationType
 
     case object UbuntuDockerImageWithEsFromApt extends EsInstallationType
   }
 
-  lazy val esDir: Path = os.root / "usr" / "share" / "elasticsearch"
-
   trait Plugin {
-    def updateEsImage(image: DockerImageDescription, config: Config): DockerImageDescription
+    def esUpdateSteps(config: Config): EsUpdateSteps
 
     def updateEsConfigBuilder(builder: EsConfigBuilder): EsConfigBuilder
 
     def updateEsJavaOptsBuilder(builder: EsJavaOptsBuilder): EsJavaOptsBuilder
   }
 
+  object Plugin {
+    final case class EsUpdateSteps(steps: List[EsUpdateStep]) {
```

Comment:
In order to execute the preexisting testcontainer update steps on WIndows native ES, I had to separate the definition of those update steps from their execution. `EsUpdateSteps` and related classes are definitions of update steps, that are then executed either on testcontainer image, or Windows ES directory

Reply:
Or maybe better: installation & configuration steps?